src/components/Dropdown/Dropdown.js

```diff
@@ -35,6 +41,35 @@ export class Dropdown extends React.Component {
     this.toggle();
   };
 
+  onTextChange = (e) => {
+    const value = e.target.value;
+    let updatedState = {
+      ...this.state,
+      isOpen: true,
+      selection: value,
+      suggestions: []
+    };
+    if (value.length > 0) {
+      const regex = new RegExp(`^${value}`, 'i');
+      const filteredSuggestions = [...this.state.options]
+        .sort()
+        .filter((v) => regex.test(v));
```

Comment:
const filteredSuggestions = this.state.options
       .filter((v) => regex.test(v))
        .sort()
        in this way you don't need to worry about modifying the original array